### PR TITLE
Dev wd

### PR DESCRIPTION
**Pull request by WardDeb**
Contains all the changes to move to github actions. Runtime is  < 7mins - linux, bit longer for macOSX ~= 10 mins

in WGBS env the r-ggplot2 version requirement has been dropped. 

Note that the azure requirements in master branch need to be dropped in this repo settings, develop is stripped already.

### CODE REVIEW

**Comment by WardDeb** [comment]
OSX envs passing now as well.

File: .github/snakePipesEnvCI.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: snakePipes_CI
+channels:
+  - conda-forge
+  - bioconda
+dependencies:
+  - python=3.9
+  - snakemake=6.2.1
+  - fuzzywuzzy
+  - mock
+  - sphinx
```
Comment (katsikora) [inline .github/snakePipesEnvCI.yml:10]: I would like to fix the sphinx version to 4.2.0, matching the specs under the docs folder, which I recently updated. Would this be possible?
And `sphinx_rtd_theme==1.0.0` .

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
+name: linux
+
+on: 
+  push:
+    branches: [dev_wd]
```
Comment (katsikora) [inline .github/workflows/linux.yml:5]: does this have to be updated to include other private branches, develop and master?

File: .github/workflows/osx.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+name: osx
+
+on:
+  push:
+    branches: [dev_wd]
```
Comment (katsikora) [inline .github/workflows/osx.yml:5]: does this have to be updated to include other private branches, develop and master?

File: .github/snakePipesEnvCI.yml (hunk 1 above)
Comment (WardDeb) [reply .github/snakePipesEnvCI.yml]: Done in 870d381

File: .github/workflows/linux.yml (hunk 2 above)
Comment (WardDeb) [reply .github/workflows/linux.yml]: The on push just makes sure tests are run when a specific branch is, which was handy to develop them. For us the 'on pullrequest' is more important. I removed on push in 870d381

File: .github/workflows/osx.yml (hunk 3 above)
Comment (WardDeb) [reply .github/workflows/osx.yml]: see comment linux.yml

**Pull request merged**